src/frequently-requested-changes.md

```diff
@@ -0,0 +1,165 @@
+proposals to add "alternative" syntaxes, in addition to those that replace the
+existing syntax. Many of these proposals come from people who also write other
+languages. Arguments range from the ergonomic ("I don't want to type this") to
+the aesthetic ("I don't like how this looks").
+
+Changes that would break existing Rust code are non-starters. Even in an
+edition, changes this fundamental remain extremely unlikely. The established
+Rust community with knowledge of existing Rust syntax has a great deal of
+value, and to be considered, a syntax change proposal would have to be not just
+*better*, but *so wildly better* as to overcome the massive downside of
+switching.
+
+In addition, such changes often go against one or more other aspects of Rust's
+design philosophy. For instance, we don't want to make changes that make code
+easier to write but harder to read, or changes that make code more error-prone
+to modify and maintain.
+
+That said, *we are open to proposals that involve new syntax*, especially for
+new features, or to improve an existing fundamental feature. The bar for new
+syntax (e.g. new operators) is high, but not insurmountable. But the bar for
+*changes to existing syntax* is even higher.
+
+## Changes to avoid writing `self.method()` when calling a method from a method
+
+In Rust, within a method of an object, calling another method requires writing
+`self.othermethod()`, and accessing a field requires `self.field`. In some
+other languages, such accesses can omit the equivalent of `self.`, and just
+write `othermethod` or `field`, implicitly referencing the "current object".
+People writing code with such method calls or field accesses sometimes ask for
+such shorthand in Rust.
+
+Rust prefers the explicitness of writing `self.` for method calls and field
+accesses, and in most ways treating `self` as a normal object of the type. This
+avoids unexpected calls to the wrong method, makes it easier to distinguish
+methods and free functions, and makes code easier to read. We don't want to
+provide a shorthand that makes this syntax briefer at the expense of this
+clarity and unambiguity.
+
+## Arbitrary custom operator syntax
+
+Rust allows overloading existing operators; for instance, you can implement the
+`Add` trait to overload the `+` operator. However, Rust does not allow creating
+*new* operators.
+
+Some would argue this can make code more readable, if you know what the
+operators mean. If you don't, it makes the code inscrutable.
+
+In general, such a change would substantially raise barriers to entry for Rust
+developers, making Rust code less approachable and less universally
+understandable. We're unlikely to add support for this.
+
+Note that Rust's existing operator overloading is semantic (`impl Add`), not
+syntactic (`impl Plus`), which tends to encourage using overloaded operators
+for the same semantic purposes, rather than for building arbitrary domain-specific languages.
```

Comment:
I don't think the distinction between "semantic"/"syntactic" is terribly clear here.

Overall, my mental model is that the overloading **is** syntactic, so maybe that is why I'm confused by the statement.

E.g., is your intent to say that the language/compiler is going to assume certain properties like commutativity/associativity for implementations of `Add`?

(It also may lead some to some wrong impressions. E.g. overloading `Add` does not automatically get one `AddAssign`, nor vice versa.)

Reply:
@pnkfelix I was attempting to make the point that our naming uses things like "Add" and "Mul" rather than "Plus"/`+` or "Star"/`*`, which encourages people to use those operators for their intended purposes rather than for arbitrary DSLs. Happy to change the language to make that clearer.